extension/agenthealth/extension.go

```diff
@@ -24,11 +25,30 @@ var _ awsmiddleware.Extension = (*agentHealth)(nil)
 func (ah *agentHealth) Handlers() ([]awsmiddleware.RequestHandler, []awsmiddleware.ResponseHandler) {
 	var responseHandlers []awsmiddleware.ResponseHandler
 	requestHandlers := []awsmiddleware.RequestHandler{useragent.NewHandler(ah.cfg.IsUsageDataEnabled)}
-	if ah.cfg.IsUsageDataEnabled {
-		req, res := stats.NewHandlers(ah.logger, ah.cfg.Stats)
-		requestHandlers = append(requestHandlers, req...)
-		responseHandlers = append(responseHandlers, res...)
+
+	if ah.cfg == nil {
+		return nil, nil
+	}
+
+	if !ah.cfg.IsUsageDataEnabled {
+		ah.logger.Debug("Usage data is disabled, skipping stats handlers")
+		return requestHandlers, responseHandlers
 	}
+
+	statusCodeEnabled := ah.cfg.IsStatusCodeEnabled
+
+	var statsResponseHandlers []awsmiddleware.ResponseHandler
+	var statsRequestHandlers []awsmiddleware.RequestHandler
+
+	if ah.cfg.Stats != nil {
+		statsRequestHandlers, statsResponseHandlers = stats.NewHandlers(ah.logger, *ah.cfg.Stats, statusCodeEnabled, true)
+	} else {
+		statsRequestHandlers, statsResponseHandlers = stats.NewHandlers(ah.logger, agent.StatsConfig{}, statusCodeEnabled, false)
+	}
```

Comment:
nit: It's a bit strange that the stats handler constructor is still being called when the stats configuration is not configured.